Implementation for set_entry_timer extern.

komaljai opened the pull request:
- Implementaion of set_entry_timer extern
- Remove error 'key field larger than previous key, move it before previous key to avoid padding between these keys in EBPF'
- Fix ebpf_packetoffsetbits_save variable declaration.
- Replace NoAction in switch case 0

komaljai committed:
Implementation for set_entry_timer.

komaljai committed:
Added testcase for set_entry_timer

syogender commented on backends/tc/backend.cpp:
```diff
@@ -446,14 +447,27 @@ void ConvertToBackendIR::postorder(const IR::P4Table *t) {
     }
 }
 
+const IR::P4Table *ConvertToBackendIR::getTableForAction(cstring action) const {
```
what will happed is same action is there in multiple table?

There is no way to find action of a table?

komaljai replied:
It is a valid issue, changed the implementation to get table_name. Getting tableName from ActionTranslationVisitorPNA class, This holds the current table's action being processed.

psivanup commented on backends/tc/backend.cpp:
```diff
@@ -446,14 +447,27 @@ void ConvertToBackendIR::postorder(const IR::P4Table *t) {
 }
 
+const IR::P4Table *ConvertToBackendIR::getTableForAction(cstring action) const {
+    for (auto entry : actionsPerTable) {
```
If one action is used in multiple tables, then it is duplicated by frontend to make it unique.
So, referring name.originalName will be incorrect. Using toString might be fine as it uses externalName() from Declaration class, please check.

komaljai replied:
Changed the approch to get table_name

psivanup reviewed (approved):
Looks fine

komaljai committed:
Addressed comments

komaljai committed:
Merge branch 'main' of https://github.com/komaljai/p4c into set_entry_timer_p4tc

komaljai committed:
Remove error 'key field larger than previous key, move it before previous key to avoid padding between these keys in EBPF'
Fix ebpf_packetoffsetbits_save variable declaration.
Replace NoAction in switch case 0

komaljai committed:
Fix cpplint issue
